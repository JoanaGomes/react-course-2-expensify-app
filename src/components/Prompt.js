import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

export const Prompt = ({ visible, onCancel, title, content, options = [] }) => {
  const extraButtons = options.map((opt) => (
    <button key={ opt.label } className='button' id={ opt.id } onClick={ opt.onClick }>{ opt.label }</button>
  ));

  return (
    <Modal
      isOpen={ visible }
      onRequestClose={ onCancel }
      closeTimeoutMS={ 200 }
      className='modal'
      contentLabel='prompt'>
      <div className='modal__top'>
        <h3 className='modal__title'>{ title }</h3>
      </div>
      <div className='modal__body'>
        <p>{ content }</p>
      </div>
      <div className='modal_buttons'>
        { extraButtons }
      </div>
    </Modal>
  )
};

Prompt.propTypes = {
  visible:    PropTypes.bool.isRequired,
  onCancel:   PropTypes.func.isRequired,
  title:      PropTypes.string.isRequired,
  content:    PropTypes.string.isRequired,
  options:    PropTypes.array
}

export default Prompt;