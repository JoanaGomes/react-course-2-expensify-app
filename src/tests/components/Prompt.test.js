import React from 'react';
import { shallow } from 'enzyme';
import Prompt from '../../components/Prompt';

test('should render Prompt correctly', () => {
  const wrapper = shallow(<Prompt visible={ true } onCancel={() => {}} title='prompt title' content='test prompt' />);
  expect(wrapper).toMatchSnapshot();
});