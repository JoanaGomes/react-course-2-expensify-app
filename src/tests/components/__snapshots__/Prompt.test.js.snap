// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Prompt correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="prompt"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__top"
  >
    <h3
      className="modal__title"
    >
      prompt title
    </h3>
  </div>
  <div
    className="modal__body"
  >
    <p>
      test prompt
    </p>
  </div>
  <div
    className="modal_buttons"
  />
</Modal>
`;

exports[`should render Prompt with a single button 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="prompt"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__top"
  >
    <h3
      className="modal__title"
    >
      prompt title
    </h3>
  </div>
  <div
    className="modal__body"
  >
    <p>
      test prompt
    </p>
  </div>
  <div
    className="modal_buttons"
  >
    <button
      className="button"
      id="prompt-btn"
      key="button"
      onClick={[Function]}
    >
      button
    </button>
  </div>
</Modal>
`;
